haystack_experimental/components/tools/openapi/_openapi.py

```diff
@@ -0,0 +1,557 @@
+            if op_id in op.operation_dict.get("operationId", ""):
+                return self.get_operation_item(path, path_item, method)
+        raise ValueError(
+            f"No operation found with operationId {op_id}, method {method}"
+        )
+
+    def get_operation_item(
+        self, path: str, path_item: Dict[str, Any], method: Optional[str] = None
+    ) -> Operation:
+        """
+        Gets a particular Operation item from the OpenAPI specification given the path and method.
+
+        :param path: The path of the operation.
+        :param path_item: The path item from the OpenAPI specification.
+        :param method: The HTTP method of the operation.
+        :return: The operation
+        """
+        if method:
+            operation_dict = path_item.get(method.lower(), {})
+            if not operation_dict:
+                raise ValueError(
+                    f"No operation found for method {method} at path {path}"
+                )
+            return Operation(path, method.lower(), operation_dict, self.spec_dict)
+        if len(path_item) == 1:
+            method, operation_dict = next(iter(path_item.items()))
+            return Operation(path, method, operation_dict, self.spec_dict)
+        if len(path_item) > 1:
+            raise ValueError(
+                f"Multiple operations found at path {path}, method parameter is required."
+            )
+        raise ValueError(f"No operations found at path {path} and method {method}")
+
+    def get_security_schemes(self) -> Dict[str, Dict[str, Any]]:
+        """
+        Get the security schemes from the OpenAPI specification.
+
+        :return: The security schemes as a dictionary.
+        """
+        return self.spec_dict.get("components", {}).get("securitySchemes", {})
+
+
+class ClientConfiguration:
+    """Configuration for the OpenAPI client."""
+
+    def __init__(  # noqa: PLR0913 pylint: disable=too-many-arguments
+        self,
+        openapi_spec: Union[str, Path],
+        credentials: Optional[str] = None,
+        request_sender: Optional[Callable[[Dict[str, Any]], Dict[str, Any]]] = None,
+        llm_provider: Optional[LLMProvider] = None,
+    ):  # noqa: PLR0913
+        if isinstance(openapi_spec, (str, Path)) and os.path.isfile(openapi_spec):
+            self.openapi_spec = OpenAPISpecification.from_file(openapi_spec)
+        elif isinstance(openapi_spec, str):
+            if is_valid_http_url(openapi_spec):
+                self.openapi_spec = OpenAPISpecification.from_url(openapi_spec)
+            else:
+                self.openapi_spec = OpenAPISpecification.from_str(openapi_spec)
+        else:
+            raise ValueError(
+                "Invalid OpenAPI specification format. Expected file path or dictionary."
+            )
+
+        self.credentials = credentials
+        self.request_sender = request_sender or send_request
+        self.llm_provider: LLMProvider = llm_provider or LLMProvider.OPENAI
```

Comment:
This should be a default value.

Reply:
@shadeMe addressed with [4f55c96](https://github.com/deepset-ai/haystack-experimental/pull/8/commits/4f55c96606af473060c6af46fb64c3537a264d7d)